azure/services/privatedns/privatedns.go

```diff
@@ -65,22 +65,24 @@ func (s *Service) Reconcile(ctx context.Context) error {
 		}
 		s.Scope.V(2).Info("successfully created private DNS zone", "private dns zone", zoneSpec.ZoneName)
 
-		// Link the virtual network.
-		s.Scope.V(2).Info("creating a virtual network link", "virtual network", zoneSpec.VNetName, "private dns zone", zoneSpec.ZoneName)
-		link := privatedns.VirtualNetworkLink{
-			VirtualNetworkLinkProperties: &privatedns.VirtualNetworkLinkProperties{
-				VirtualNetwork: &privatedns.SubResource{
-					ID: to.StringPtr(azure.VNetID(s.Scope.SubscriptionID(), zoneSpec.VNetResourceGroup, zoneSpec.VNetName)),
+		for _, linkSpec := range zoneSpec.Links {
+			// Link each virtual network.
+			s.Scope.V(2).Info("creating a virtual network link", "virtual network", linkSpec.VNetName, "private dns zone", zoneSpec.ZoneName)
+			link := privatedns.VirtualNetworkLink{
+				VirtualNetworkLinkProperties: &privatedns.VirtualNetworkLinkProperties{
+					VirtualNetwork: &privatedns.SubResource{
+						ID: to.StringPtr(azure.VNetID(s.Scope.SubscriptionID(), linkSpec.VNetResourceGroup, linkSpec.VNetName)),
+					},
+					RegistrationEnabled: to.BoolPtr(false),
 				},
-				RegistrationEnabled: to.BoolPtr(false),
-			},
-			Location: to.StringPtr(azure.Global),
-		}
-		err = s.client.CreateOrUpdateLink(ctx, s.Scope.ResourceGroup(), zoneSpec.ZoneName, zoneSpec.LinkName, link)
-		if err != nil {
-			return errors.Wrapf(err, "failed to create virtual network link %s", zoneSpec.LinkName)
+				Location: to.StringPtr(azure.Global),
+			}
+			err = s.client.CreateOrUpdateLink(ctx, s.Scope.ResourceGroup(), zoneSpec.ZoneName, linkSpec.LinkName, link)
```

Comment:
It doesn't look like we are checking if the link already exists. If it does, it would be beneficial to verify the state matches the desired state and not doing another PUT. We should only do the PUT if the desired state does not match the actual state.